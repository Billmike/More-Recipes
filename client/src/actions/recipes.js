import uuid from 'uuid';

export const addRecipe = ({ name = '', description = '', img_url = 'no-img-here', category = '', ingredients = [], instructions = [] } = {}) => ({
  type: 'ADD_RECIPE',
  recipe: {
    id: uuid(),
    name,
    description,
    img_url,
    category,
    ingredients,
    instructions,
  },
});

export const editRecipe = (id, updates) => ({
  type: 'EDIT_RECIPE',
  id,
  updates,
});

export const removeRecipe = ({ id } = {}) => ({
  type: 'REMOVE_RECIPE',
  id
})
