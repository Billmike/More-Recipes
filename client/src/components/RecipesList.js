import React from 'react';
import pizza from '../assets/img/pizzza.jpg';

const RecipeList = ({ name, description, category }) => {
	return (
		<div className="card-group">
        <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
          <div className="card index-card">
            <a href="./template/recipe_detail.html"><img className="card-img-top" src={ pizza } alt="Card image cap" /></a>
            <div className="card-body">
              <a href="./template/recipe_detail.html"><h4 className="card-title">{name}</h4></a>
              <p className="card-text">{description}</p>
              <a href="./template/recipe_detail.html" className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></a>
              <a href="#" className="btn border border-secondary rounded" data-toggle="modal" data-target="#exampleModal">  89 <i className="fa fa-heart" aria-hidden="true" /></a>
              <a href="./template/recipe_detail.html#comment" className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></a>
            </div>
            <div className="card-footer">
              <small className="text-muted">Recipe Category - {category}</small>
            </div>
          </div>
      </div>
		)
};

export default RecipeList;
