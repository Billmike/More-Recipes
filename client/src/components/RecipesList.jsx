import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const RecipeList = ({ id, name, description, category }) => (
  <div className="card-group">
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
      <div className="card index-card">
        <Link to={`/recipe/${id}`}><img className="card-img-top" alt="Pizza" /></Link>
        <div className="card-body">
          <Link to={`/recipe/${id}`}><h4 className="card-title">{name}</h4></Link>
          <p className="card-text">{description}</p>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">45 <i className="fa fa-eye" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded">  89 <i className="fa fa-heart" aria-hidden="true" /></Link>
          <Link to={`/recipe/${id}`} className="btn border border-secondary rounded"> 6 <i className="fa fa-comments" aria-hidden="true" /></Link>
        </div>
        <div className="card-footer">
          <small className="text-muted">Recipe Category - {category}</small>
        </div>
      </div>
  </div>
);

export default connect()(RecipeList);
