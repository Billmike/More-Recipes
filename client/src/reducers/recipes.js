const recipeDefaultState = [];

export default (state = recipeDefaultState, action) => {
  switch (action.type) {
    case 'ADD_RECIPE':
      return [
        ...state,
        action.recipe,
      ];
    default:
      return state;
  }
};
