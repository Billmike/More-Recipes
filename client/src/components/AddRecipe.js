import React from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

const AddRecipe = (props) => {
    console.log(props)
	return (
	   <div>
       <h1 className="container">Add Recipe</h1>
        <RecipesForm onSubmit={(recipe) => {
            props.dispatch(addRecipe(recipe));
            props.history.push('/')
        }} />
       </div>	
	)
}

export default connect()(AddRecipe);
