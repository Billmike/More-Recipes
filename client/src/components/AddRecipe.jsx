import React, { Component } from 'react';
import RecipesForm from './RecipesForm';
import { connect } from 'react-redux';
import { addRecipe } from '../actions/recipes';

export class AddRecipe extends Component {
  onSubmit = (recipe) => {
    this.props.addRecipe(recipe);
    this.props.history.push('/dashboard');
  };

  render() {
    return (
      <div>
        <h1 className="container"> Add Recipe </h1>
        <RecipesForm
          onSubmit={this.onSubmit}
        />
      </div>
      );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addRecipe: (recipe) => dispatch(addRecipe(recipe))
  }
}

// const AddRecipe = (props) => {
//     console.log(props)
// 	return (
// 	   <div>
//        <h1 className="container">Add Recipe</h1>
//         <RecipesForm onSubmit={(recipe) => {
//             props.dispatch(addRecipe(recipe));
//             props.history.push('/')
//         }} />
//        </div>	
// 	)
// }

export default connect(undefined, mapDispatchToProps)(AddRecipe);
