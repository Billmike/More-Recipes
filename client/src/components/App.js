import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import Favorite from './Favorite';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import NotFoundPage from './NotFoundPage';

class App extends Component {
    render() {
        return(
            <div>
                <BrowserRouter>
                    <div>
                        <Navbar />
                        <Switch>
                        <Route exact path="/" component={Homepage} />
                        <Route path="/dashboard" component={Dashboard} />
                        <Route path="/favorite" component={Favorite} />
                        <Route path="/add" component={AddRecipe} />
                        <Route path="/edit" component={EditRecipe} />
                        <Route path="/profile" component={ProfilePage} />
                        <Route component={NotFoundPage} />
                    </Switch>
                    </div>
                </BrowserRouter>
            </div>
        )
    }
}

export default App;
