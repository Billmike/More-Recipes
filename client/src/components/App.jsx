import React from 'react';
import { BrowserRouter, browserHistory, Route, Switch } from 'react-router-dom';
import Navbar from './Navbar';
import Homepage from './Hompage';
import Dashboard from './Dashboard';
import RecipeDetails from './RecipeDetails';
import Favorite from './Favorites';
import AddRecipe from './AddRecipe';
import EditRecipe from './EditRecipe';
import ProfilePage from './ProfilePage';
import LoginPage from './LoginPage';
import RegistrationPage from './RegistrationPage';
import RequireAuth from '../utils/RequireAuth';
import NotFoundPage from './NotFoundPage';

const App = () => (
  <div>
    <BrowserRouter history={browserHistory}>
      <div>
        <Navbar />
        <Switch>
          <Route exact path="/" component={Homepage} />
          <Route path="/dashboard" component={RequireAuth(Dashboard)} />
          <Route path="/favorites" component={RequireAuth(Favorite)} />
          <Route path="/add" component={RequireAuth(AddRecipe)} />
          <Route path="/edit/:id" component={RequireAuth(EditRecipe)} />
          <Route path="/recipe/:id" component={RecipeDetails} />
          <Route path="/profile" component={RequireAuth(ProfilePage)} />
          <Route path="/login" component={LoginPage} />
          <Route path="/register" component={RegistrationPage} />
          <Route component={NotFoundPage} />
        </Switch>
      </div>
    </BrowserRouter>
  </div>
);

export default App;
