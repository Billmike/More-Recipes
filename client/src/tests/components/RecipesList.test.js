import React from 'react';
import { shallow } from 'enzyme';
import { RecipeList } from '../../components/RecipesList';
import recipes from '../fixtures/recipes';

test('Should display the list of recipes in the application', () => {
	const wrapper = shallow(<RecipeList {...recipes[0]} />);
	expect(wrapper).toMatchSnapshot();
});
