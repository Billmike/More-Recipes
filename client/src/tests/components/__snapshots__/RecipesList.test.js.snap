// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display the list of recipes in the application 1`] = `
<div
  className="card-group"
>
  <div
    className="card index-card"
  >
    <Link
      replace={false}
      to="/recipe/1"
    >
      <img
        alt="Pizza"
        className="card-img-top"
      />
    </Link>
    <div
      className="card-body"
    >
      <Link
        replace={false}
        to="/recipe/1"
      >
        <h4
          className="card-title"
        >
          Fried Rice
        </h4>
      </Link>
      <p
        className="card-text"
      >
        Awesome fried rice
      </p>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
        45 
        <i
          aria-hidden="true"
          className="fa fa-eye"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
          89 
        <i
          aria-hidden="true"
          className="fa fa-heart"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
         6 
        <i
          aria-hidden="true"
          className="fa fa-comments"
        />
      </Link>
    </div>
    <div
      className="card-footer"
    >
      <small
        className="text-muted"
      >
        Recipe Category - 
        Lunch
      </small>
    </div>
  </div>
  <div
    className="card index-card"
  >
    <Link
      replace={false}
      to="/recipe/1"
    >
      <img
        alt="Pizza"
        className="card-img-top"
      />
    </Link>
    <div
      className="card-body"
    >
      <Link
        replace={false}
        to="/recipe/1"
      >
        <h4
          className="card-title"
        >
          Fried Rice
        </h4>
      </Link>
      <p
        className="card-text"
      >
        Awesome fried rice
      </p>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
        45 
        <i
          aria-hidden="true"
          className="fa fa-eye"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
          89 
        <i
          aria-hidden="true"
          className="fa fa-heart"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
         6 
        <i
          aria-hidden="true"
          className="fa fa-comments"
        />
      </Link>
    </div>
    <div
      className="card-footer"
    >
      <small
        className="text-muted"
      >
        Recipe Category - 
        Lunch
      </small>
    </div>
  </div>
  <div
    className="card index-card"
  >
    <Link
      replace={false}
      to="/recipe/1"
    >
      <img
        alt="Pizza"
        className="card-img-top"
      />
    </Link>
    <div
      className="card-body"
    >
      <Link
        replace={false}
        to="/recipe/1"
      >
        <h4
          className="card-title"
        >
          Fried Rice
        </h4>
      </Link>
      <p
        className="card-text"
      >
        Awesome fried rice
      </p>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
        45 
        <i
          aria-hidden="true"
          className="fa fa-eye"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
          89 
        <i
          aria-hidden="true"
          className="fa fa-heart"
        />
      </Link>
      <Link
        className="btn border border-secondary rounded"
        replace={false}
        to="/recipe/1"
      >
         6 
        <i
          aria-hidden="true"
          className="fa fa-comments"
        />
      </Link>
    </div>
    <div
      className="card-footer"
    >
      <small
        className="text-muted"
      >
        Recipe Category - 
        Lunch
      </small>
    </div>
  </div>
</div>
`;
